backend/routes/coach.ts

```diff
@@ -110,40 +108,9 @@ export async function deleteCoach(
         .then((parsed) => util.isAdmin(parsed))
         .then((parsed) => util.mutable(parsed, parsed.data.id))
         .then(async (parsed) => {
-            return ormL
-                .searchLoginUserByPerson(parsed.data.id)
-                .then((logUs) => {
-                    if (
-                        logUs !== null &&
-                        logUs.login_user_id !== parsed.userId
-                    ) {
-                        return removeAllKeysForLoginUserId(logUs.login_user_id)
-                            .then(() => {
-                                return Promise.resolve({});
-                            })
-                            .then(() => {
-                                return ormL
-                                    .deleteLoginUserByPersonId(parsed.data.id)
-                                    .then(() => {
-                                        return ormP
-                                            .deletePersonById(parsed.data.id)
-                                            .then(() => Promise.resolve({}))
-                                            .catch(() =>
-                                                Promise.reject(
-                                                    errors.cookServerError()
-                                                )
-                                            );
-                                    })
-                                    .catch(() =>
-                                        Promise.reject(errors.cookServerError())
-                                    );
-                            })
-                            .catch(() =>
-                                Promise.reject(errors.cookServerError())
-                            );
-                    }
-                    return Promise.reject(errors.cookInvalidID());
-                });
+            return ormLU
+                .deleteLoginUserFromDB(parsed.data.id)
+                .then(() => Promise.resolve({}));
```

Comment:
Wouldn't an admin that is also a coach be able to delete himself?

Reply:
nope! I've tested this and an admin that is also a coach can't delete himself.